### PR TITLE
event: Auto-generate event definitions

### PR DESCRIPTION
**Pull request by hajimehoshi**
As event package needs to depend on driver package, and drvier
package needs to emit events, we need event definitions in both
side. This change add a generator for event definitions.

Updates #926

### CODE REVIEW

**Comment by hajimehoshi** [comment]
@beoran Could you take a look? Thanks!

File: event/event.go [hunk 1]
```diff
@@ -40,46 +32,64 @@ type KeyboardKeyCharacter struct {
...
 type GamepadAxis struct {
 	// ID represents which gamepad caused the event.
 	ID int
+
 	// Axis is the axis of the game pad that changed position.
 	Axis int
-	// Position is the psoition of the axis after the change.
-	// It varies between -1.0 and 1.0.
+
+	// Position is the psoition of the axis after the change. It varies between -1.0 and 1.0.
```
Comment (beoran) [inline event/event.go]: psosition -> position

File: genevents.go [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+	return strings.Split(e.Comment, " ")[0]
+}
+
+type Member struct {
+	Comment string
+	Type    string
+}
+
+func (m *Member) Name() string {
+	return strings.Split(m.Comment, " ")[0]
```
Comment (beoran) [inline genevents.go]: `return strings.SplitN(m.Comment, " ", 2)[0]`
will be more efficient.

File: genevents.go [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+					Comment: "Height is the height of the view. This value is expressed in device independent pixels.",
+					Type:    "int",
+				},
+			},
+		},
+	}
+)
+
+const (
+	license = `// Copyright 2013 The Ebiten Authors
```
Comment (beoran) [inline genevents.go]: I think legally this should be 2013-2019 to cover the whole development range. But I am not a lawyer.

File: event/event.go (hunk 1 above)
Comment (hajimehoshi) [reply event/event.go]: Done.

File: genevents.go (hunk 2 above)
Comment (hajimehoshi) [reply genevents.go]: Good point. Done.

File: genevents.go (hunk 3 above)
Comment (hajimehoshi) [reply genevents.go]: As far as I know, only one year when the file is created should be fine (e.g., Chromium does so). The problem is that auto-generated file cannot determine the year, then I adopted 2013 which Ebiten started.

**Comment by hajimehoshi** [comment]
Thank you!

**Pull request merged**